scripts/pushsafer.pl

```diff
@@ -0,0 +1,318 @@
+    Irssi::print('From '.$caller[3].': '.$text);
+}
+
+sub send_push {
+    my $private_key = Irssi::settings_get_str('pushsafer_key');
+    if (!$private_key) {
+        debug('Missing Pushsafer.com private or alias_key.');
+        return;
+    }
+
+    debug('Sending notification.');
+    my ($channel, $text) = @_;
+    my $resp = LWP::UserAgent->new()->post(
+        'https://www.pushsafer.com/api', [
+            k => $private_key,
+            m => $text,
+            d => Irssi::settings_get_str('pushsafer_device'),
+            s => Irssi::settings_get_str('pushsafer_sound'),
+            i => Irssi::settings_get_str('pushsafer_icon'),
+            v => Irssi::settings_get_str('pushsafer_vibration'),
+            u => Irssi::settings_get_str('pushsafer_url'),
+            ut => Irssi::settings_get_str('pushsafer_urltitle'),
+            l => Irssi::settings_get_str('pushsafer_time2live'),
+            t => $channel
+        ]
+    );
+
+    if ($resp->is_success) {
+        debug('Notification successfully sent.');
+    }
+    else {
+        debug('Notification not sent: '.$resp->decoded_content);
+    }
+}
+
+sub msg_pub {
+    my ($server, $data, $nick, $address, $target) = @_;
+    my $safeNick = quotemeta($server->{nick});
+
+    if(check_ignore_channels($target)) {
+        return;
+    }
+
+    if(check_ignore($address) || check_away($server)) {
+        return;
+    }
+
+    if ($data =~ /$safeNick/i) {
+        debug('Got pub msg.');
+        send_push($target, $nick.': '.strip_formating($data));
+    }
+}
+
+sub msg_print_text {
+    my ($dest, $text, $stripped) = @_;
+    my $server = $dest->{server};
+    my $target = $dest->{target};
+
+    return if (!$server || !($dest->{level} & MSGLEVEL_HILIGHT));
+
+    if(check_ignore_channels($target)) {
+        return;
+    }
+
+    if(check_away($server)) {
+        return;
+    }
+
+    debug('Got nick highlight');
+    $stripped =~ s/^\s+|\s+$//g;
+    send_push($target, $stripped);
+}
+
+sub msg_pri {
+    my ($server, $data, $nick, $address) = @_;
+
+    if(check_ignore($address) || check_away($server)) {
+        return;
+    }
+    debug('Got priv msg.');
+    send_push('Priv, '.$nick, strip_formating($data));
+}
+
+sub msg_kick {
+    my ($server, $channel, $nick, $kicker, $address, $reason) = @_;
+
+    if(check_ignore($address) || check_away($server)) {
+        return;
+    }
+
+    if ($nick eq $server->{nick}) {
+        debug('Was kicked.');
+        send_push('Kicked: '.$channel, 'Was kicked by: '.$kicker.'. Reason: '.strip_formating($reason));
+    }
+}
+
+sub msg_test {
+   my ($data, $server, $item) = @_;
+   $data =~ s/^([\s]+).*$/$1/;
+   my $orig_debug = Irssi::settings_get_bool('pushsafer_debug');
+   Irssi::settings_set_bool('pushsafer_debug', 1);
+   debug("Sending test message :" . $data);
+   send_push("Test Message", strip_formating($data));
+   Irssi::settings_set_bool('pushsafer_debug', $orig_debug);
+}
+
+sub strip_formating {
```

Comment:
please test if Irssi::strip_codes works as a replacement for your custom function?